MealProblem.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+   public class MealProblem
+    {
+    // coll is an array for use in tie cases.
+    public static int[] coll = null;
+    // exceptional_case_item is used for the case when in all parameter tie occur
+    public static int exceptional_case_item = -1;
+
+    // This method is just choosing the maximum content value item when CAPITAL alphabet given as 'T' or anyother
+    public static int ChooseMax(int[] participent_element)
+       {
+           try{
+        int count=0;
+        int max = -1;
+         for(int i=0;i<coll.Length;i++)
+          if(coll[i]==0)
+           {
+              max = i;
+              break; 
+           }
+        for(int i=1;i<participent_element.Length;i++)
+         {
+          if(participent_element[i]>participent_element[max]&&coll[i]==0)
+           {
+             max = i;
+           }
+         }
+         for(int i=0;i<participent_element.Length;i++)
+         {
+          if(participent_element[i]==participent_element[max])
+            count++;
+          else 
+            coll[i] = 1;
+         }
+        if(count==1)
+         return max;
+        exceptional_case_item = max;
+           }
+           catch(Exception e)
+            {
+             Console.WriteLine(e);
+            }
+            return -1;
+       }
+
+// This method is just choosing the minimum content value item when SMALL alphabet given as 't' or anyother
+       public static int ChooseMin(int[] participent_element)
+       {
+        try{
+        int count=0;
+        int min = -1;
+        for(int i=0;i<coll.Length;i++)
+          if(coll[i]==0)
+           {
+              min = i;
+              break; 
+           }
+        for(int i=1;i<participent_element.Length;i++)
+         {
+          if(participent_element[i]<participent_element[min]&&coll[i]==0)
+           {
+             min = i;
+           }
+         }
+         for(int i=0;i<participent_element.Length;i++)
+          {if(participent_element[i]==participent_element[min])
+            count++;
+            else
+            coll[i] = 1;
+          }
+        if(count==1)
+         return min;
+        exceptional_case_item = min;
+        
+        }
+        catch(Exception e)
+         {
+            Console.WriteLine(e);
+         }
+         return -1;
+       }
+
+       // This is the prime calculation method
+      static public int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
```

Comment:
Test scenarios, present in Program.cs are failing with NullReference exception. 
Please call it from program.cs and test all scenarios.